osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySliderBall.cs

```diff
@@ -1,64 +1,77 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using osu.Framework.Allocation;
 using osu.Framework.Bindables;
+using osu.Framework.Extensions.ObjectExtensions;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
 using osu.Framework.Graphics.Sprites;
 using osu.Game.Rulesets.Objects.Drawables;
 using osu.Game.Rulesets.Osu.Objects.Drawables;
 using osu.Game.Skinning;
+using osuTK;
 using osuTK.Graphics;
 
 namespace osu.Game.Rulesets.Osu.Skinning.Legacy
 {
     public partial class LegacySliderBall : CompositeDrawable
     {
-        private readonly Drawable animationContent;
-
         private readonly ISkin skin;
 
         [Resolved(canBeNull: true)]
         private DrawableHitObject? parentObject { get; set; }
 
-        public Color4 BallColour => animationContent.Colour;
-
         private Sprite layerNd = null!;
+        private Drawable ball = null!;
         private Sprite layerSpec = null!;
 
-        public LegacySliderBall(Drawable animationContent, ISkin skin)
+        public Color4 BallColour => ball.Colour;
+
+        public LegacySliderBall(ISkin skin)
         {
-            this.animationContent = animationContent;
             this.skin = skin;
 
             AutoSizeAxes = Axes.Both;
         }
 
         [BackgroundDependencyLoader]
-        private void load()
+        private void load(DrawableHitObject drawableHitObject)
         {
             var ballColour = skin.GetConfig<OsuSkinColour, Color4>(OsuSkinColour.SliderBall)?.Value ?? Color4.White;
 
+            DrawableSlider drawableSlider = (DrawableSlider)drawableHitObject;
+
+            // stable apparently calculates slider velocity in units of seconds rather than milliseconds.
+            double stableSliderVelocity = drawableSlider.HitObject.Velocity * 1000;
```

Comment:
This has been briefly discussed over at [discord](https://discord.com/channels/188630481301012481/188630652340404224/1165006192645525524). The gist of it is that stable apparently calculates slider velocity as if it's path distance travelled per second, meanwhile lazer calculates it on a per millisecond basis. Stable still transfers it back to milliseconds when calculating duration in [here](https://github.com/peppy/osu-stable-reference/blob/3ea48705eb67172c430371dcfc8a16a002ed0d3d/osu!/GameplayElements/HitObjects/Osu/SliderOsu.cs#L853), notice multiplication of 1000 while dividing by velocity).